### PR TITLE
Move Kubelet Summary API to staging directory

### LINKED ISSUE #8: Add kubelet stats API to k8s.io/kubelet
I would be really useful to move kubelet Summary API to k8s.io/kubelet

Kubelet Summary API is used as part of Core Metrics Pipeline and imported by https://github.com/kubernetes-sigs/metrics-server

### PR DESCRIPTION
**Pull request by serathius**
This PR moves Kubelet stats summary to kubelet staging directory, making it easy importable via "k8s.io/kubelet".
Without this PR projects that want to consume this API need to import "k8s.io/kubernetes" which is doesn't work with golang modules. 

Fixes https://github.com/kubernetes/kubelet/issues/8

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
Related PRs (for approvers to review):
- https://github.com/kubernetes/kubernetes/pull/92632
- https://github.com/kubernetes/kubernetes/pull/92527

File: pkg/kubelet/apis/BUILD [hunk 1]
```diff
@@ -38,7 +38,6 @@ filegroup(
         "//pkg/kubelet/apis/config:all-srcs",
         "//pkg/kubelet/apis/podresources:all-srcs",
         "//pkg/kubelet/apis/resourcemetrics/v1alpha1:all-srcs",
-        "//pkg/kubelet/apis/stats/v1alpha1:all-srcs",
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/apis/BUILD:41]: Also update: https://github.com/kubernetes/kubernetes/blob/01f98c4321cd03e15f16d6d4ea42797f2d4bca3c/hack/.import-aliases#L51

**Comment by SergeyKanzhelev** [comment]
I am curious is it really fixes this issue? 

> Fixes kubernetes-sigs/metrics-server#372

Was the issue cause by incompatibility?

**Comment by liggitt** [comment]
/hold

@dashpole is this the API surfaced by `--enable-cadvisor-json-endpoints`? If so, according to https://github.com/kubernetes/kubernetes/blob/b826d394af56d1bc77aebee356ec1622ea450fbe/cmd/kubelet/app/options/options.go#L365-L367 and https://github.com/kubernetes/kubernetes/issues/68522 this will be removed in 1.20. Shouldn't we avoid exporting the types for an alpha API that is being removed?

**Comment by serathius** [comment]
This PR moves Summary API, which is not part of cAdvisor JSON endpoints, but a K8s own API. It's not under `--enable-cadvisor-json-endpoints`. 
Still Summary API is planned for deprecation ` [TBD] Deprecate the Summary API and cAdvisor prometheus endoints by adding and deprecating a --enable-container-monitoring-endpoints flag`, but timeline is unknown as we need to propose `out-of-tree` replacement.

**Comment by dashpole** [comment]
@serathius is correct.  This is not in the scope of --enable-cadvisor-json-endpoints, and we have yet to propose a replacement.

**Comment by liggitt** [comment]
ok, thanks

/hold cancel

**Comment by SergeyKanzhelev** [comment]
@serathius please update the file I mentioned in this comment: https://github.com/kubernetes/kubernetes/pull/92985#discussion_r454663078

**Comment by serathius** [comment]
@SergeyKanzhelev done
/assign @bentheelder @derekwaynecarr

**Comment by max-rocket-internet** [comment]
@serathius are you sure this fixes my issue that you linked to?

**Comment by serathius** [comment]
> Marek Siarkowicz are you sure this fixes my issue that you linked to?

Sorry, I linked wrong Issue, correct one is https://github.com/kubernetes/kubelet/issues/8

**Comment by max-rocket-internet** [comment]
> Sorry, I linked wrong Issue

Damn it 😐

**Comment by SergeyKanzhelev** [comment]
@serathius would you please rebase? easy merge once rebased - has all the milestone and labels

**Comment by SergeyKanzhelev** [comment]
test failures says to run:

```
hack/update-vendor.sh
```

**Comment by SergeyKanzhelev** [comment]
/hold

to prevent automatic re-tests

**Comment by SergeyKanzhelev** [comment]
Now verify wants `Run ./hack/update-gofmt.sh`

**Pull request merged**